tiledb/sm/query/test/unit_query_condition.cc

```diff
@@ -2350,50 +3634,10 @@ TEST_CASE(
   }
   REQUIRE(tile->write(values, 0, cells * sizeof(uint64_t)).ok());
 
-  // Build a combined query for `> 3 AND <= 6`.
-  uint64_t cmp_value_1 = 3;
-  QueryCondition query_condition_1;
-  REQUIRE(query_condition_1
-              .init(
-                  std::string(field_name),
-                  &cmp_value_1,
-                  sizeof(uint64_t),
-                  QueryConditionOp::GT)
-              .ok());
-  // Run Check
-  REQUIRE(query_condition_1.check(array_schema).ok());
-  uint64_t cmp_value_2 = 6;
-  QueryCondition query_condition_2;
-  REQUIRE(query_condition_2
-              .init(
-                  std::string(field_name),
-                  &cmp_value_2,
-                  sizeof(uint64_t),
-                  QueryConditionOp::LE)
-              .ok());
-  // Run Check
-  REQUIRE(query_condition_2.check(array_schema).ok());
-  QueryCondition query_condition_3;
-  REQUIRE(query_condition_1
-              .combine(
-                  query_condition_2,
-                  QueryConditionCombinationOp::AND,
-                  &query_condition_3)
-              .ok());
-
-  // Apply the query condition.
-  uint64_t cell_count = 0;
-  std::vector<uint8_t> result_bitmap(cells, 1);
-  REQUIRE(query_condition_3
-              .apply_sparse<uint8_t>(
-                  array_schema, result_tile, result_bitmap, &cell_count)
-              .ok());
-
-  // Check that the cell slab now contains cell indexes 4, 5, and 6.
-  REQUIRE(cell_count == 3);
-  for (uint64_t cell_idx = 0; cell_idx < cells; ++cell_idx) {
-    REQUIRE(
-        result_bitmap[cell_idx] == (cell_idx >= 4 && cell_idx <= 6 ? 1 : 0));
+  std::vector<TestParams> tp_vec;
+  populate_test_params_vector(field_name, &result_tile, tp_vec);
+  for (auto& elem : tp_vec) {
+    validate_qc(elem, cells, array_schema, result_tile);
```

Comment:
This looks great now!! One small change... make validate_qc, validate_qc_sparse and validate_qc_dense.

Then this for loop becomes:
SECTION("Validate apply") {
  for (auto& elem : tp_vec) {
    validate_qc(elem, cells, array_schema, result_tile);
  }
}

SECTION("Validate apply sparse") {
  for (auto& elem : tp_vec) {
    validate_qc_sparse(elem, cells, array_schema, result_tile);
  }
}

SECTION("Validate apply dense") {
  for (auto& elem : tp_vec) {
    validate_qc_dense(elem, cells, array_schema, result_tile);
  }
}

Reply:
Resolved in PR!